internal/reconciler/logpipeline/sync.go

```diff
@@ -197,6 +199,12 @@ func (s *syncer) syncTLSConfigSecret(ctx context.Context, logPipelines []telemet
 		if err = controllerutil.SetOwnerReference(&logPipelines[i], &newSecret, s.Scheme()); err != nil {
 			return fmt.Errorf("unable to set owner reference for tls config secret: %w", err)
 		}
+
+		if tls.Cert.IsDefined() && tls.Key.IsDefined() {
```

Comment:
Maybe let's change the control flow here similar to the Otel case.

Reply:
done